### ISSUE TITLE
refcount decremented with record scalar (migrated from Trac #1269)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2822]
Original ticket http://projects.scipy.org/numpy/ticket/1269
Reported 2009-10-21 by trac user batripler, assigned to unknown.

The following function reduces the refcount of None (by one) every time it is run. If the dtype is expanded to include more objects, the refcount will be decremented by one for each additional object field added to the record.

```
import sys,numpy
def numpybug():
  print sys.getrefcount(None)
  dt=numpy.dtype([('a',object),('b',object)])
  sc=numpy.empty(1, dt)[0]; a=numpy.empty(1, dt)
  a[:]=sc
  del sc, a
  print sys.getrefcount(None)
```

**Comment by thouis** [comment]
Comment in Trac by trac user batripler, 2010-04-29

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2010-09-05

Seems to have been fixed since 1.3.0 --- the bug is not present in current trunk.
